src/silent-blocks/silent-blocks.gateway.ts

```diff
@@ -0,0 +1,37 @@
+import { Injectable, Logger } from '@nestjs/common';
+import {
+    WebSocketGateway,
+    WebSocketServer,
+    OnGatewayConnection,
+    OnGatewayDisconnect,
+} from '@nestjs/websockets';
+import { Server, WebSocket } from 'ws';
+
+@Injectable()
+@WebSocketGateway()
+export class SilentBlocksGateway
+    implements OnGatewayConnection, OnGatewayDisconnect
+{
+    private readonly logger = new Logger(SilentBlocksGateway.name);
+
+    @WebSocketServer() server: Server;
+
+    handleConnection(client: WebSocket) {
+        const remoteAddress = (client as any)._socket.remoteAddress;
+        this.logger.debug(`Client connected: ${remoteAddress}`);
+    }
+
+    handleDisconnect(client: WebSocket) {
+        const remoteAddress = (client as any)._socket.remoteAddress;
+        this.logger.debug(`Client disconnected: ${remoteAddress}`);
+    }
+
+    // Method to broadcast silent block to all connected clients
+    broadcastSilentBlock(silentBlock: Buffer) {
+        this.server.clients.forEach((client: WebSocket) => {
+            if (client.readyState === WebSocket.OPEN) {
+                client.send(silentBlock.toString('hex')); // Send silent block as hex string
+            }
+        });
```

Comment:
1. because everything inside `forEach` is a synchronous operation, we should use a simple `for..of` loop. `forEach` is very slow.
2. I see that you are sending the block as a hex string. This breaks the pattern we use for REST endpoints. We send the block as `Buffer` in REST endpoints. I think we should stick to using `Buffer` here as well for consistency.